### PR TITLE
view.rst: first-pass revision (with `Kokkos_Views.hpp` in hand)

### PR DESCRIPTION
**Pull request by ajpowelsnl**
Changes herein represent a "first pass," "side-by-side" with the code (on `develop`) revision, editing for completeness (parameters, Type defs, Public Class Members) and making minor improvements.

These changes were made with the recently-discussed [API standard entry](https://github.com/kokkos/internal-documents/blob/master/documentation_sessions/2023/doc_session_may_23_2023.md) in mind.

Among the **TODO**:

- More definitions for more of the typedefs, and details on how and when to use them
- Clarify usage scenarios for "NonMember" Functions
- Review constructors
- Review data layout, dimensions and strides
- Review Assignment Rules
- Review other (is there a better placement for these items?)

### CODE REVIEW

File: docs/source/API/core/view/view.rst [hunk 1]
```diff
@@ -6,18 +6,18 @@
...
+Kokkos View is a potentially reference counted, multi-dimensional array (up to eight dimensions), with compile-time layouts and memory space.
 Its semantics are similar to that of ``std::shared_ptr``.
 
 Interface
 ---------
 
 .. code-block:: cpp
 
-    template <class DataType [, class LayoutType] [, class MemorySpace] [, class MemoryTraits]>
+    template <class DataType [, class Space (required)] [, class ArrayLayout (optional)] [, class MemoryTraits(optional)]>
```
Comment (masterleinad) [inline docs/source/API/core/view/view.rst:20]:
```suggestion
    template <class DataType [, class Space] [, class Layout] [, class MemoryTraits]>
```
`Space` is not required, "[" implies optional.

File: docs/source/API/core/view/view.rst [hunk 2]
```diff
@@ -36,30 +36,31 @@ Parameters
 .. |LayoutStride| replace:: :cppkokkos:func:`LayoutStride`
 
 Template parameters other than ``DataType`` are optional, but ordering is enforced.
-That means for example that ``LayoutType`` can be omitted but if both ``MemorySpace``
+That means for example that ``Layout`` can be omitted, but if both ``MemorySpace``
```
Comment (masterleinad) [inline docs/source/API/core/view/view.rst:39]:
```suggestion
That means for example that ``ArrayLayout`` can be omitted, but if both ``MemorySpace``
```
or keep it as `LayoutType` or `Layout` but it needs to be consistent.

File: docs/source/API/core/view/view.rst [hunk 3]
```diff
@@ -36,30 +36,31 @@ Parameters
...
 and ``MemoryTraits`` are specified, ``MemorySpace`` must come before ``MemoryTraits``.
 
 * ``DataType``:
 
   Defines the fundamental scalar type of the ``View`` and its dimensionality.
-  The basic structure is ``ScalarType STARS BRACKETS`` where the number of STARS denotes
-  the number of runtime length dimensions and the number of BRACKETS defines the compile time dimensions.
-  Due to C++ type restrictions runtime dimensions must come first.
+  The basic structure is ``ScalarType number of STARS, number of BRACKETED items`` where the number of STARS denotes
```
Comment (masterleinad) [inline docs/source/API/core/view/view.rst:45]: It's not relly the number of STARS here but really the STARS themselves, right? I liked the previous version better.

File: docs/source/API/core/view/view.rst [hunk 4]
```diff
@@ -36,30 +36,31 @@ Parameters
...
+  the number of runtime length dimensions, while the number of BRACKETED items defines the compile-time dimensions.
+  Due to C++ type restrictions runtime dimensions must come first, and runtime dimensions cannot be interspersed with compile-time dimensions.
```
Comment (masterleinad) [inline docs/source/API/core/view/view.rst:47]:
```suggestion
  Due to C++ type restrictions, runtime dimensions must come first, and runtime dimensions cannot be interspersed with compile-time dimensions.
```

File: docs/source/API/core/view/view.rst [hunk 5]
```diff
@@ -36,30 +36,31 @@ Parameters
...
   - ``const int***[5][3]``: 5D View of ``int`` with 3 runtime and 2 compile dimensions. The data is ``const``.
 
-  - ``Foo[6][2]``: 2D View of a class ``Foo`` with 2 compile time dimensions.
+  - ``Foo[6][2]``: 2D View of a class instance ``Foo`` with 2 compile time dimensions.
 
-* ``LayoutType``:
+* ``Layout``:
 
   Determines the mapping of indices into the underlying 1D memory storage.
+  If no layout is given, it can be derived from the memory space associated with the execution space.
```
Comment (masterleinad) [inline docs/source/API/core/view/view.rst:59]:
```suggestion
  If no layout is given, the default layout for the memory space associated with the execution space is used.
```

File: docs/source/API/core/view/view.rst [hunk 6]
```diff
@@ -36,30 +36,31 @@ Parameters
...
+  the number of runtime length dimensions, while the number of BRACKETED items defines the compile-time dimensions.
+  Due to C++ type restrictions runtime dimensions must come first, and runtime dimensions cannot be interspersed with compile-time dimensions.
   Examples:
 
   - ``double**``: 2D View of ``double`` with 2 runtime dimensions
 
   - ``const int***[5][3]``: 5D View of ``int`` with 3 runtime and 2 compile dimensions. The data is ``const``.
 
-  - ``Foo[6][2]``: 2D View of a class ``Foo`` with 2 compile time dimensions.
+  - ``Foo[6][2]``: 2D View of a class instance ``Foo`` with 2 compile time dimensions.
```
Comment (masterleinad) [inline docs/source/API/core/view/view.rst:54]: `Foo` really is the class, not the instance.
```suggestion
  - ``Foo[6][2]``: 2D View of type ``Foo`` with 2 compile time dimensions.
```

File: docs/source/API/core/view/view.rst [hunk 7]
```diff
@@ -68,24 +69,24 @@ and ``MemoryTraits`` are specified, ``MemorySpace`` must come before ``MemoryTra
 
 * ``MemorySpace``:
 
-  Controls the storage location.
-  If omitted the default memory space of the default execution space is used (i.e. ``Kokkos::DefaultExecutionSpace::memory_space``)
+  Kokkos Views store data in Memory Spaces.
```
Comment (masterleinad) [inline docs/source/API/core/view/view.rst:72]: Do we really want to mention/duplicate `Kokkos` all the time?

File: docs/source/API/core/view/view.rst [hunk 8]
```diff
@@ -68,24 +69,24 @@ and ``MemoryTraits`` are specified, ``MemorySpace`` must come before ``MemoryTra
...
 
 .. _Atomic: ../atomics.html
 
 .. |Atomic| replace:: :cppkokkos:func:`Atomic`
 
 
 * ``MemoryTraits``:
-
-  Sets access properties via enum parameters for the templated ``Kokkos::MemoryTraits<>`` class.
+  Memory Traits specify how a data structure is accessed in an algorithm.
```
Comment (masterleinad) [inline docs/source/API/core/view/view.rst:81]:
```suggestion
  MemoryTraits specify how a data structure is accessed in an algorithm.
```
or
```suggestion
  memory traits specify how a data structure is accessed in an algorithm.
```
?

File: docs/source/API/core/view/view.rst (hunk 3 above)
Comment (ajpowelsnl) [reply docs/source/API/core/view/view.rst:45]: Correct : )

**Pull request closed without merging**